gprofiler/metadata/application_identifiers.py

```diff
@@ -0,0 +1,248 @@
+            if arg_key == arg_name:
+                return arg_val
+
+    return _NON_AVAILABLE_ARG
+
+
+def _get_cli_arg_by_index(args: List[str], index: int) -> str:
+    try:
+        return args[index]
+    except IndexError:
+        return _NON_AVAILABLE_ARG
+
+
+def _append_python_module_to_proc_wd(process: Process, module_path: str) -> str:
+    if os.path.isabs(module_path):
+        return module_path
+
+    return os.path.join(process.cwd(), module_path)
+
+
+class _GunicornApplicationIdentifier(_ApplicationIdentifier):
+    def get_application_name(self, process: Process) -> Optional[str]:
+        # As of gunicorn documentation the WSGI module name most probably will come from the cmdline and not from the
+        # config file / environment variables (they added the option to specify `wsgi_app`
+        # in the config file only in version 20.1.0)
+
+        if "gunicorn" not in _get_cli_arg_by_index(process.cmdline(), 0) or "gunicorn" not in _get_cli_arg_by_index(
+            process.cmdline(), 1
+        ):
+            return None
+
+        for arg in process.cmdline():
+            # ":" might appear in ip:port or in wsgi app specification (module:func)
+            if ":" in arg:
+                if _IP_PORT_RE.match(arg):
+                    continue
+
+                return f"gunicorn-{_append_python_module_to_proc_wd(process, arg)}"
+
+        _logger.warning(
+            f"GunicornApplicationSeparator: matched against process {process} but couldn't find WSGI module"
+        )
+        return None
+
+
+class _UwsgiApplicationIdentifier(_ApplicationIdentifier):
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        if "uwsgi" not in _get_cli_arg_by_index(process.cmdline(), 0):
+            return None
+
+        wsgi_arg = _get_cli_arg_by_name(process.cmdline(), "-w")
+        if wsgi_arg is not _NON_AVAILABLE_ARG:
+            return f"uwsgi-{_append_python_module_to_proc_wd(process, wsgi_arg)}"
+
+        wsgi_config = self._find_wsgi_from_config_file(process)
+        if wsgi_config is not None:
+            return f"uwsgi-{_append_python_module_to_proc_wd(process, wsgi_config)}"
+
+        _logger.warning(
+            "Couldn't find uwsgi wsgi module, both from cmdline and from config file",
```

Comment:
In `_GunicornApplicationIdentifier` you prefix your log message with the class name; here you don't. Be consistent.

Reply:
I removed the log in `_GunicornApplicationIdentifier` so not relevant anymore.